### DISCUSSION TITLE
PrimeBlocks and PrimeVue V4 - Limited Component Integration?

### DISCUSSION THREAD

**Discussion by dsherley9** [primefaces/community#2689] [PrimeVue Blocks] [answered]
I recently purchased PrimeBlocks and was expecting more seamless integration with PrimeVue V4 components, particularly when it comes to utilizing design tokens, pass-through properties for different design strategies, and slots for customizing content within blocks. While the design and structure of PrimeBlocks are visually appealing, it seems more Tailwind-based than PrimeVue-based, which adds complexity when reusing components and maintaining a DRY (Don’t Repeat Yourself) approach.

Are there any plans to make PrimeBlocks more compatible with PrimeVue’s core features, such as leveraging built-in components, slots, and design tokens for easier customization? A tighter integration would significantly reduce the effort required to reuse elements and maintain consistency within PrimeVue projects.

**Comment by dsherley9** [comment]
I’ve noticed that in some cases, PrimeVue components aren't being used where they could potentially add value. For example, in the Cards with Progress Bar section on [PrimeBlocks Stats](https://primeblocks.org/application/stats), the progress bars are custom-built using Tailwind classes. Wouldn't it be more valuable to leverage PrimeVue’s own [ProgressBar component](https://primevue.org/progressbar/) here?

For instance, PrimeVue’s component page provides examples with state variables like this:

```vue
<template>
    <div class="card">
        <ProgressBar :value="40"> {{ value }}/100 </ProgressBar>
    </div>
</template>

<script setup>
import { ref } from "vue";

const value = ref(40);
</script>
```

Pairing multiple PrimeVue V4 components to build a PrimeBlock, using the component library, and adding customization with Tailwind where necessary, seems like a natural fit for users who are already invested in the PrimeVue ecosystem and expect tighter integration.

**Comment by cagataycivici** [comment, marked as answer]
Idea behind blocks is to go beyond what the open source core offers, for example core has a Tabs component but Blocks offer more [options](https://primeblocks.org/application/tabs).  There is a Drawer and Menu in core but Sidebar Layouts offer [more](https://primeblocks.org/application/sidebar-layout). There is a progressbar but the block you mentioned demonstrates how you can build one yourself to get full customization and control. 

I think we found the right balance overall, Blocks use the color design tokens only so that they can be used both with Styled and Unstyled mode. If we used the component tokens from the styled mode, they would be incompatible with Unstyled mode.

Thank you for your feedback, we'll still keep improving and use more components where necessary in upcoming updates.

**Reply by dsherley9** [reply]
Thank you for the thoughtful reply and for explaining the reasoning behind PrimeBlocks. I understand the goal of offering more options and control, and it’s great to hear that there's an emphasis on customization and flexibility.

I appreciate the balance you’re striving for with Styled and Unstyled modes. It’s also reassuring to know that you’re open to incorporating more PrimeVue components in future updates where it makes sense. As someone deeply invested in the PrimeVue ecosystem, I’m excited to see how PrimeBlocks continues to evolve.

Thanks again for considering my feedback, and I look forward to the improvements!
